File: Python/Elektronik.py
# File: Elektronik.py

from Produk import Produk

class Elektronik(Produk):
    def __init__(self, kode_produk="", nama_produk="", merek="", harga=0, tahun_produksi=0, status="",
                 garansi=0, warna="", material=""):
        super().__init__(kode_produk, nama_produk, merek, harga, tahun_produksi, status)
        self._garansi = garansi
        self._warna = warna
        self._material = material

    # Getter methods
    def get_garansi(self): return self._garansi
    def get_warna(self): return self._warna
    def get_material(self): return self._material

    # Setter methods
    def set_garansi(self, garansi): self._garansi = garansi
    def set_warna(self, warna): self._warna = warna
    def set_material(self, material): self._material = material

File: Java/Elektronik.java
// File: Elektronik.java

public class Elektronik extends Produk {
    private int garansi;
    private String warna;
    private String material;

    public Elektronik(String kodeProduk, String namaProduk, String merek, int harga, int tahunProduksi, String status,
                      int garansi, String warna, String material) {
        super(kodeProduk, namaProduk, merek, harga, tahunProduksi, status);
        this.garansi = garansi;
        this.warna = warna;
        this.material = material;
    }

    // Getter methods
    public int getGaransi() { return this.garansi; }
    public String getWarna() { return this.warna; }
    public String getMaterial() { return this.material; }

    // Setter methods
    public void setGaransi(int garansi) { this.garansi = garansi; }
    public void setWarna(String warna) { this.warna = warna; }
    public void setMaterial(String material) { this.material = material; }
}